assimilation_code/programs/preprocess/preprocess.f90

```diff
@@ -65,450 +120,453 @@ program preprocess
 !character(len = 78) :: blank_comment_line = &
 !'!                                                                           '
 
+integer, parameter :: NUM_SECTIONS = 7
 ! List of the DART PREPROCESS strings for obs_def type files.
-character(len = 29) :: preprocess_string(8) = (/ &
+character(len = 29) :: preprocess_string(NUM_SECTIONS) = (/ &
       'MODULE CODE                  ', &
       'USE FOR OBS_QTY_MOD          ', &
       'USE OF SPECIAL OBS_DEF MODULE', &
       'GET_EXPECTED_OBS_FROM_DEF    ', &
       'READ_OBS_DEF                 ', &
       'WRITE_OBS_DEF                ', &
-      'INTERACTIVE_OBS_DEF          ', &
-      'THE EIGHTH ONE IS UNDEFINED  '/)
+      'INTERACTIVE_OBS_DEF          '/)
 
 !! Must match the list above.  Is there a default code section that we can
 !! autogenerate for this section?
-!logical :: default_code(8) = &
+!logical :: default_code(NUM_SECTIONS) = &
 !   (/ .false., &
 !      .false., &
 !      .false., &
 !      .true.,  &
 !      .true.,  &
 !      .true.,  &
-!      .true.,  &
-!      .false. /)
-
-integer, parameter :: module_item = 1
-integer, parameter :: kind_item = 2
-integer, parameter :: use_item = 3
-integer, parameter :: get_expected_item = 4
-integer, parameter :: read_item = 5
-integer, parameter :: write_item = 6
-integer, parameter :: interactive_item = 7
-
-integer :: num_input_files = 0
-integer :: num_model_files = 0
+!      .true.  /)
+
+integer, parameter :: MODULE_ITEM = 1
+integer, parameter :: QTY_ITEM = 2
+integer, parameter :: USE_ITEM = 3
+integer, parameter :: GET_EXPECTED_ITEM = 4
+integer, parameter :: READ_ITEM = 5
+integer, parameter :: WRITE_ITEM = 6
+integer, parameter :: INTERACTIVE_ITEM = 7
+
+integer :: num_obs_type_files = 0
+integer :: num_quantity_files = 0
 integer :: obs_def_in_unit, obs_def_out_unit
-integer :: obs_kind_in_unit, obs_kind_out_unit, in_unit
+integer :: obs_qty_in_unit, obs_qty_out_unit, in_unit
+
+integer, parameter   :: MAX_OBS_TYPE_FILES = 1000
+integer, parameter   :: MAX_QUANTITY_FILES = 1000
+logical :: file_has_usercode(MAX_OBS_TYPE_FILES) = .false.
 
-integer, parameter   :: max_input_files = 1000
-integer, parameter   :: max_model_files = 1000
-logical :: file_has_usercode(max_input_files) = .false.
+
+integer, parameter :: NML_STRLEN = 256
```

Comment:
i believe this is right, but confirm?  i think we set the max length of a pathname/filename in the system to be 256.  we set message strings to be 512 so we can include a filename plus supporting error text without running into string-too-long crashes.

Reply:
confirmed with Tim in person review.